src/persist-client/src/impl/state.rs

```diff
@@ -316,6 +332,20 @@ where
     }
 }
 
+// Test helpers.
+#[cfg(test)]
+impl<K, V, T, D> State<K, V, T, D>
+where
+    K: Codec,
+    V: Codec,
+    T: Timestamp + Lattice + Codec64,
+    D: Codec64,
+{
+    pub fn num_trace_batches(&self) -> usize {
```

Comment:
I dislike this helper. maybe I can move the test to mod state?

Reply:
I think the test can be in `mod state`, because the optimization only affects state itself.